backends/ebpf/runtime/ebpf_runtime_test.h

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+
+ */
+
+#ifndef BACKENDS_EBPF_RUNTIME_EBPF_RUNTIME_TEST_H_
+#define BACKENDS_EBPF_RUNTIME_EBPF_RUNTIME_TEST_H_
+
+#include <unistd.h>     // getopt()
+#include <ctype.h>      // isprint()
+#include <string.h>     // memcpy()
+#include <stdlib.h>     // malloc()
+#include "test.h"
+#include "pcap_util.h"
+
+/**
+ * @brief Feed a list packets into an eBPF program.
+ * @details This is a mock function emulating the behavior of a running
+ * eBPF program. It takes a list of input packets and iteratively parses them
+ * using the given imported ebpf_filter function. The output defines whether
+ * or not the packet is "dropped." If the packet is not dropped, its content is
+ * copied and appended to an output packet list.
+ *
+ * @param pkt_list A list of input packets running through the filter.
+ * @return The list of packets "surviving" the filter function
+ */
+static pcap_list_t *feed_packets(pcap_list_t *pkt_list, int debug) {
```

Comment:
You should probably pass ebpf_filter as a function pointer.
Have a `typedef int (*packet_filter)(skb* s)`
And make feed_packets have a parameter of type packet_filter.